### PR TITLE
Various post 4.7 items: Planner notes, salinity, code cleanup

### PR DESCRIPTION
**Pull request by sfuchs79**
### Describe the pull request:

- [ ] Bug fix
- [x] Functional change
- [ ] New feature
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

Three post v4.7 items I collected while working on more important v4.7 stuff. Hold them back to not generate too much noise. Now that we are not in a hurry any more...

Planner notes: Remove some unnecessary double info (string!)

Salinity: Init salinity for new diveplan with value from displayed dive

Cylinder handling: Non functional change, code cleanup. Myself responsible for part of the old lines and while reviewing recognized that this became too ugly too understand it.

### CODE REVIEW

**Comment by dirkhh** [comment]
@atdotde would you take a look at the first commit, please? I no longer pretend to understand that code.

@sfuchs79 did you mean "Tidy up"? I can't comment on commit messages (GitHub flaw). I'd love it if you could fix that and push again. Do you know how to do that?
`git rebase -i HEAD~3`
then replace the 'pick' in front of the commit where you want to reword the message with an 'r' and save the file you are shown.
You'll get an editor to reword the message. Change, safe, all done.

File: qt-models/cylindermodel.cpp [hunk 1]
```diff
@@ -233,11 +233,8 @@ QVariant CylindersModel::data(const QModelIndex &index, int role) const
 		if (index.column() == REMOVE) {
 			same_gas = same_gasmix_cylinder(cyl, index.row(), &displayed_dive, false);
 
-			if ((in_planner() || same_gas == -1) &&
-				((DivePlannerPointsModel::instance()->currentMode() != DivePlannerPointsModel::NOTHING &&
```
Comment (atdotde) [inline qt-models/cylindermodel.cpp:237]: I have to admit, I am a bit lost. What was this currentMode() != ...:NOTHING and why is it not necessary anymore? A slightly more verbose commit message would be helpful.

**Review by atdotde — COMMENTED** [review]
Why not go the full way? Even 0 bar / 0 Liter sounds redundant. I would leave out that cylinder completely or rather say something like “unused”.

**Review by atdotde — COMMENTED** [review]
Good catch.

File: qt-models/cylindermodel.cpp (hunk 1 above)
Comment (sfuchs79) [reply qt-models/cylindermodel.cpp:237]: currentMode() != ...:NOTHING is IMHO equivalent to in_planner() but in_planner() is much more familiar to everyone.
This is why I replaced it.

**Comment by sfuchs79** [comment]
@atdotde Hmmm...

922ℓ/39bar von EAN32 (42ℓ/2bar im berechneten Aufstieg)
 — Minimum Gas (basierend auf 3.3xAMV/+4min@15m): 735ℓ/30bar/Δ:+122bar
Unbenutzt EAN50
234ℓ/32bar von Sauerstoff (234ℓ/31bar im berechneten Aufstieg)
Unbenutzt Luft

vs.

922ℓ/39bar von EAN32 (42ℓ/2bar im berechneten Aufstieg)
 — Minimum Gas (basierend auf 3.3xAMV/+4min@15m): 735ℓ/30bar/Δ:+122bar
0ℓ/0bar EAN50
234ℓ/32bar von Sauerstoff (234ℓ/31bar im berechneten Aufstieg)
0ℓ/0bar Luft

I prefer the second look. Consistent layout...

@dirkhh Need a spell checker even for the commit messages. Will fix it.

**Comment by dirkhh** [comment]
So what's the status on this one? Is it ready to merge?
@atdotde - I usually just leave those to you :-)

**Pull request merged**